api/src/main/java/org/pmiops/workbench/workspaces/VwbWorkspaceServiceImpl.java

```diff
@@ -0,0 +1,229 @@
+import org.pmiops.workbench.rawls.model.RawlsWorkspaceDetails;
+import org.pmiops.workbench.tanagra.model.Cohort;
+import org.pmiops.workbench.tanagra.model.CohortList;
+import org.pmiops.workbench.tanagra.model.FeatureSet;
+import org.pmiops.workbench.tanagra.model.FeatureSetList;
+import org.pmiops.workbench.utils.mappers.FirecloudMapper;
+import org.pmiops.workbench.utils.mappers.WorkspaceMapper;
+import org.pmiops.workbench.wsm.WsmClient;
+import org.pmiops.workbench.wsmanager.ApiException;
+import org.pmiops.workbench.wsmanager.model.IamRole;
+import org.pmiops.workbench.wsmanager.model.WorkspaceDescription;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+@Service(VwbWorkspaceServiceImpl.VWB_WORKSPACE_SERVICE)
+public class VwbWorkspaceServiceImpl implements WorkspaceService {
+
+  private static final Logger logger = LoggerFactory.getLogger(VwbWorkspaceServiceImpl.class);
+
+  public static final String VWB_WORKSPACE_SERVICE = "vwbWorkspaceService";
+
+  private final WsmClient wsmClient;
+
+  private final WorkspaceMapper workspaceMapper;
+
+  private final FirecloudMapper firecloudMapper;
+
+  private final WorkspaceDao workspaceDao;
+
+  private final InitialCreditsService expirationService;
+
+  private final WorkspaceAuthService workspaceAuthService;
+
+  public VwbWorkspaceServiceImpl(
+      WsmClient wsmClient,
+      WorkspaceMapper workspaceMapper,
+      FirecloudMapper firecloudMapper,
+      WorkspaceDao workspaceDao,
+      InitialCreditsService expirationService,
+      WorkspaceAuthService workspaceAuthService) {
+    this.wsmClient = wsmClient;
+    this.workspaceMapper = workspaceMapper;
+    this.firecloudMapper = firecloudMapper;
+    this.workspaceDao = workspaceDao;
+    this.expirationService = expirationService;
+    this.workspaceAuthService = workspaceAuthService;
+  }
+
+  @Override
+  public WorkspaceResponse getWorkspace(String workspaceNamespace, String workspaceId) {
+
+    DbWorkspace dbWorkspace = workspaceDao.getRequired(workspaceNamespace, workspaceId);
+    workspaceAuthService.validateWorkspaceTierAccess(dbWorkspace);
+
+    RawlsWorkspaceDetails fcWorkspace;
+    WorkspaceResponse workspaceResponse = new WorkspaceResponse();
+
+    WorkspaceDescription workspaceDescription =
+        wsmClient.getWorkspaceAsService(dbWorkspace.getWorkspaceNamespace());
+    fcWorkspace = workspaceMapper.toWorkspaceDetails(workspaceDescription);
+
+    workspaceResponse.setAccessLevel(
+        firecloudMapper.fcToApiWorkspaceAccessLevel(
+            firecloudMapper.fromIamRole(workspaceDescription.getHighestRole())));
+    Workspace workspace =
+        workspaceMapper.toApiWorkspace(dbWorkspace, fcWorkspace, expirationService);
+    workspaceResponse.setWorkspace(workspace);
+
+    return workspaceResponse;
+  }
+
+  @Override
+  public boolean notebookTransferComplete(String workspaceNamespace, String workspaceId) {
+    return false;
```

Comment:
if set to false, does clone workspace work? I think there is some check in the UI - hide transfer spinner when transfer complete? https://github.com/all-of-us/workbench/blob/c4f0957478f2dbf3c2ba15a769344a30134a4d8a/ui/src/app/pages/appAnalysis/app-files-list.tsx#L106

Also, this need to be implemented, suggest to add a ticket:
CloneWorksapceAPI in WSM have asyn and sync. One option is to use that async api then check transfer complete, and reuse existing UI code above. (can chat more in person)